strategies/iCARL/experiment.py

```diff
@@ -0,0 +1,191 @@
+import sys
+sys.path.append(r'/disk1/semola/avalanche')
+
+import unittest
+
+from os.path import expanduser
+
+import torch
+
+from avalanche.benchmarks.datasets import CIFAR100
+from avalanche.benchmarks.utils import AvalancheDataset
+from avalanche.models import IcarlNet, make_icarl_net, initialize_icarl_net
+from avalanche.training.plugins.lr_scheduling import LRSchedulerPlugin
+from torch.optim import SGD
+from torchvision import transforms
+from avalanche.benchmarks.generators import nc_benchmark
+from avalanche.training.plugins import EvaluationPlugin
+from avalanche.evaluation.metrics import ExperienceAccuracy, StreamAccuracy
+from avalanche.logging.interactive_logging import InteractiveLogger
+import random
+import numpy as np
+from torch.optim.lr_scheduler import MultiStepLR
+
+from strategies.utils import get_average_metric
+from avalanche.training.strategies.icarl import ICaRL
+
+
+def get_dataset_per_pixel_mean(dataset):
+    result = None
+    patterns_count = 0
+
+    for img_pattern, _ in dataset:
+        if result is None:
+            result = torch.zeros_like(img_pattern, dtype=torch.float)
+
+        result += img_pattern
+        patterns_count += 1
+
+    if result is None:
+        result = torch.empty(0, dtype=torch.float)
+    else:
+        result = result / patterns_count
+
+    return result
+
+
+def icarl_cifar100_augment_data(img):
+    img = img.numpy()
+    padded = np.pad(img, ((0, 0), (4, 4), (4, 4)), mode='constant')
+    random_cropped = np.zeros(img.shape, dtype=np.float32)
+    crop = np.random.randint(0, high=8 + 1, size=(2,))
+
+    # Cropping and possible flipping
+    if np.random.randint(2) > 0:
+        random_cropped[:, :, :] = \
+            padded[:, crop[0]:(crop[0]+32), crop[1]:(crop[1]+32)]
+    else:
+        random_cropped[:, :, :] = \
+            padded[:, crop[0]:(crop[0]+32), crop[1]:(crop[1]+32)][:, :, ::-1]
+    t = torch.tensor(random_cropped)
+    return t
+
+
+class Config(dict):
+    def __getattribute__(self, key):
+        try:
+            return self[key]
+        except KeyError:
+            raise AttributeError(key)
+
+    def __setattr__(self, key, value):
+        self[key] = value
+
+
+class iCARL(unittest.TestCase):
+    """
+        Reproducing iCaRL experiments from paper
+        "iCaRL: Incremental Classifier and Representation Learning",
+        Sylvestre-Alvise Rebuffi, Alexander Kolesnikov, Georg Sperl, Christoph H. Lampert;
+        Proceedings of the IEEE Conference on
+        Computer Vision and Pattern Recognition (CVPR), 2017, pp. 2001-2010
+        https://openaccess.thecvf.com/content_cvpr_2017/html/Rebuffi_iCaRL_Incremental_Classifier_CVPR_2017_paper.html
+    """
+
+    def test_iCIFAR100_batch10(self, override_args=None):
+        """
+            iCIFAR-100 with 10 batches - benchmark
+        """
+        # config
```

Comment:
can you use the `create_default_args` to build the arguments? Take a look at how it is done in the other strategies (e.g., test_pmnist in EWC).

Reply:
ok